homeassistant/components/emoncms_history.py

```diff
@@ -0,0 +1,92 @@
+"""
+A component which allows you to send data to Emoncms.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/emoncms_history/
+"""
+import logging
+from datetime import timedelta
+import json
+
+import voluptuous as vol
+import requests
+
+
+from homeassistant.const import (
+    CONF_API_KEY, CONF_WHITELIST,
+    EVENT_STATE_CHANGED, CONF_URL, STATE_UNKNOWN,
+    STATE_UNAVAILABLE)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers import state as state_helper
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "emoncms_history"
+CONF_INPUTNODE = "inputnode"
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=1)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Required(CONF_URL): cv.string,
+        vol.Required(CONF_INPUTNODE): cv.positive_int,
+        vol.Required(CONF_WHITELIST, default=[]):
+            vol.All(cv.ensure_list, [cv.entity_id]),
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Setup the emoncms_history component."""
+    conf = config[DOMAIN]
+    whitelist = conf.get(CONF_WHITELIST)
+    json_body = {}
+
+    def emoncms_event_listener(event):
+        """Listen for new messages on the bus and sends them to Dweet.io."""
+        state = event.data.get('new_state')
+        if state is None or state.state in (STATE_UNKNOWN, "",
+                                            STATE_UNAVAILABLE) \
+                or state.entity_id not in whitelist:
+            return
+
+        try:
+            _state = state_helper.state_as_number(state)
+        except ValueError:
+            return
+
+        json_body[state.entity_id] = _state
+        json_str = json.dumps(json_body, separators=(",", ":"))
```

Comment:
the input api of emoncms does not accept normal json or at least it would not work if itried it
They require data to be send like in this format for example
http://192.168.0.101:8083/emoncmsnew/input/post.json?node=22&json={input1:10,input2:10,input3:30}
so those lines are there to convert the data so it becomes a format like this : `{key1:value1,key2:value2,key3:value3}` 
it doesnt accept "key1":"value1" nor does it accept valueswith spaces after the `:` and `,`